iphone/Classes/TiUIClipboardProxy.m

```diff
@@ -339,49 +328,58 @@ - (void)setItems:(id)args
       YES);
 }
 
-- (id)getItems:(id)unused
+- (NSArray<NSDictionary<NSString *, id> *> *)getItems
 {
-  __block id items;
-
+  __block NSMutableArray<NSDictionary<NSString *, id> *> *result = [[[NSMutableArray alloc] init] retain];
   TiThreadPerformOnMainThread(
       ^{
-        items = [[[self pasteboard] items] retain];
+        NSArray<NSDictionary<NSString *, id> *> *items = [self pasteboard].items;
 
         // Check for invalid UTI's / mime-types to prevent a runtime-crash
-        for (NSDictionary *item in items) {
-          for (NSString *key in [item allKeys]) {
-            if ([key hasPrefix:@"dyn."]) {
-              NSLog(@"[ERROR] Invalid mime-type specified to setItems() before. Returning an empty result ...");
-
-              RELEASE_TO_NIL(items);
-              items = @[];
-              break;
+        for (NSDictionary<NSString *, id> *item in items) {
+          NSMutableDictionary<NSString *, id> *newItem = item.mutableCopy;
+          for (NSString *key in newItem.allKeys) {
+            if ([key isEqualToString:@"com.apple.uikit.color"]) {
+              // Convert colors back to hex strings
+              newItem[key] = [TiUtils hexColorValue:item[key]];
+            } else if (UTTypeConformsTo((CFStringRef)key, kUTTypeURL)) {
+              // Convert public.url and public.file-url values from NSURL to NSString
+              newItem[key] = [TiUtils stringValue:item[key]];
+            } else if (UTTypeConformsTo((CFStringRef)key, kUTTypeImage)) {
+              // Convert UIImage to TiBlob!
+              newItem[key] = [[[TiBlob alloc] initWithImage:(UIImage *)item[key]] autorelease];
             }
           }
-          if ([items count] == 0) {
-            break;
-          }
+          [result addObject:newItem];
         }
       },
       YES);
 
-  return [items autorelease];
+  return [result autorelease];
 }
 
-- (void)setData:(id)args
+- (void)setData:(NSString *)mimeType withData:(id)data
```

Comment:
Probably better method name can be -
```suggestion
- (void)setData:(id)data withDataType:(NSString *) mimeType
```

Reply:
It likely would be, but I'm constrained by the argument order we expose to JS (datatype, then data)